django/thunderstore/account/service_account.py

```diff
@@ -0,0 +1,109 @@
+import ulid2
+from django import forms
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+
+from thunderstore.account.models import ServiceAccount
+from thunderstore.repository.models import UploaderIdentity
+
+
+def create_service_account_username(id_: str) -> str:
+    return f"{id_}.sa@thunderstore.io"
+
+
+class CreateServiceAccountForm(forms.Form):
+    nickname = forms.CharField(max_length=32)
+
+    def __init__(self, user: User, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.user = user
+        self.fields["identity"] = forms.ModelChoiceField(
+            queryset=UploaderIdentity.objects.filter(members__user=user),
+        )
+
+    def clean_identity(self) -> UploaderIdentity:
+        identity = self.cleaned_data["identity"]
+        if not identity.can_create_service_account(self.user):
+            raise ValidationError("Must be identity owner to create a service account")
+        return identity
+
+    def clean(self) -> None:
+        super().clean()
+        nickname = self.cleaned_data.get("nickname")
+        identity = self.cleaned_data.get("identity")
+
+        if not all((nickname, identity)):
+            return
+
+        if identity.service_accounts.filter(
+            user__first_name=nickname,
+        ).exists():
+            raise ValidationError("Service account nickname must be unique")
+
+    def save(self) -> ServiceAccount:
```

Comment:
`@transaction.atomic` here also

Reply:
Alternatively could extract the service account creating logic into a function, wrap that into transaction, and call it here. Up to the implementation.